### PR TITLE
[10.0][ADD] Module which add a payment_signal

### PR DESCRIPTION
**Pull request by apesquero**
This module creates a field to enter the payment signal,
which is automatically calculated by a percentage associated with the
company.
These amounts, automatically appear in Invoice Order.

### CODE REVIEW

**Comment by apesquero** [comment]
What's wrong with the damn Readme.rst? I do not understand what is wrong, I have copied it from someone else.

**Comment by apesquero** [comment]
payment_signal/views/inherit_res_company.xml:1: [W7908(missing-newline-extrafiles), ]  Missing newline
payment_signal/views/inherit_sale_order.xml:1: [W7908(missing-newline-extrafiles), ]  Missing newline

Please, help.

In my code there are newlines. Why Travis say no!!

https://github.com/apesquero/sale-workflow/blob/10.0/payment_signal/models/inherit_res_company.py
https://github.com/apesquero/sale-workflow/blob/10.0/payment_signal/models/inherit_sale_order.py

**Comment by apesquero** [comment]
With the large amount of foolish changes I've made!
I can assure you that I am clumsy. ;)

**Comment by apesquero** [comment]
Can someone explain to me what I'm doing wrong?

Thank you.

File: payment_signal/README.rst [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+Contributors
+------------
+
+* Amaro Pesquero <apesquero@gmail.com>
+
+
+Do not contact contributors directly about support or help with technical issues.
+
+
```
Comment (JayVora-SerpentCS) [inline payment_signal/README.rst]: Kindly prepare the readme file acc to https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

File: payment_signal/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    'name': 'Payment Signal',
+    'summary': 'Module to signal quotations and orders',
+    'version': '10.0.1.0.0',
+    'author': 'Amaro Pesquero, Odoo Community Association (OCA)',
+    'license': 'AGPL-3',
+    'data': ['views/inherit_res_company.xml',
+             'views/inherit_sale_order.xml', ],
+    'category': 'Sales',
+    'depends': ['account', 'sale'],
+    'application': True,
```
Comment (JayVora-SerpentCS) [inline payment_signal/__manifest__.py]: We must set application:False here

File: payment_signal/models/inherit_res_company.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
+
+
+class ResCompany(models.Model):
+    _inherit = 'res.company'
+
+    default_signal = fields.Integer('Percent Signal in S.O.',
+                                    default=20,
+                                    store=True)
```
Comment (JayVora-SerpentCS) [inline payment_signal/models/inherit_res_company.py]: You should not use store=True not fields which are not computed, default they are stored

File: payment_signal/models/inherit_res_company.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    @api.constrains('default_signal')
```
Comment (JayVora-SerpentCS) [inline payment_signal/models/inherit_res_company.py]: use api.multi here and ensure one.

Comment (apesquero) [reply payment_signal/models/inherit_res_company.py]: I'm sorry I do not understand it. If I do not put a "constrain", you can save the field with a negative value or greater than 100. And I do not want that.

I don't understand "ensure one".

Thank you very much.

**Comment by apesquero** [comment]
Hello, testing my module, I'm seeing that in "travis" I see an error related to sale_triple_discount, which has no relation to my module.

Some clue?

**Comment by apesquero** [comment]
Forgive for my ignorance. "Needs review" means that I have to review it or that another member is going to review it.

Thank you.

File: payment_signal/README.rst [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+This module creates a field to enter the payment signal,
+which is automatically calculated by a percentage associated with the company.
+These amounts, automatically appear in Invoice Order.
+
+.. image:: static/payment-signal-01.png
+
+.. image:: static/payment-signal-02.png
+
+.. image:: static/payment-signal-03.png
+
```
Comment (JayVora-SerpentCS) [inline payment_signal/README.rst:28]: Please add runbot link

**Review by JayVora-SerpentCS — CHANGES_REQUESTED** [review]

File: payment_signal/README.rst (hunk 5 above)
Comment (apesquero) [reply payment_signal/README.rst:28]: Sorry, how can I create the route of the rumbot link? Thank you.

Comment (rousseldenis) [reply payment_signal/README.rst:28]: You can avoid managing those things creating readme directory and let readme bot do the job.
Please read https://github.com/OCA/maintainer-tools/pull/339 details

**Comment by rousseldenis** [comment]
@apesquero Do you plan to finish this ?

**Comment by rousseldenis** [comment]
Closing this as very old and no answer from author. Please feel free to reopen it if needed.

**Pull request closed without merging**